src/snowflake/snowpark/dataframe.py

```diff
@@ -1040,7 +1041,8 @@ def to_pandas(
         # this might happen when calling this method with non-select commands
         # e.g., session.sql("create ...").to_pandas()
         if block:
-            check_is_pandas_dataframe_in_to_pandas(result)
+            if not isinstance(result, pandas.DataFrame):
+                return pd.DataFrame(result)
```

Comment:
what's the context of the change? 

also can you check the connector that is the result guaranted to be a dict or something that can be put into a pandas DF?

Reply:
it is guaranteed to be data or iter or just raise error as it is guarded by the code in server_connection:
https://github.com/snowflakedb/snowpark-python/blob/47ee32c51b6ea1914ffcd6a8a541949918a18cd0/src/snowflake/snowpark/_internal/server_connection.py#L578
context is here: https://snowflake.slack.com/archives/C01BD4CSEFN/p1744278063299269